tensorflow_addons/metrics/matthews_correlation_coefficient.py

```diff
@@ -0,0 +1,152 @@
+"""Matthews Correlation Coefficient Implementation."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.keras.metrics import Metric
+import numpy as np
+
+
+class MatthewsCorrelationCoefficient(Metric):
+    """Computes the Matthews Correlation Coefficient.
+
+    The statistic is also known as the phi coefficient.
+    The Matthews correlation coefficient (MCC) is used in
+    machine learning as a measure of the quality of binary
+    and multiclass classifications. It takes into account
+    true and false positives and negatives and is generally
+    regarded as a balanced measure which can be used even
+    if the classes are of very different sizes. The correlation
+    coefficient value of MCC is between -1 and +1. A
+    coefficient of +1 represents a perfect prediction,
+    0 an average random prediction and -1 an inverse
+    prediction. The statistic is also known as
+    the phi coefficient.
+
+    MCC = (TP * TN) - (FP * FN) /
+          ((TP + FP) * (TP + FN) * (TN + FP ) * (TN + FN))^(1/2)
+
+    Args:
+       num_classes : Number of unique classes in the dataset.
+
+    Returns:
+       mcc : float (the Matthews correlation coefficient)
+
+    Usage:
+    ```python
+    actuals = tf.constant([[1.0], [1.0], [1.0], [0.0]],
+             dtype=tf.float32)
+    preds = tf.constant([[1.0], [0.0], [1.0], [1.0]],
+             dtype=tf.float32)
+    # Matthews correlation coefficient
+    mcc = MatthewsCorrelationCoefficient(num_classes=1)
+    output.update_state(actuals, preds)
+    print('Matthews correlation coefficient is:',
+    output.result().numpy())
+    # Matthews correlation coefficient is : -0.33333334
+
+    ```
+    """
+
+    def __init__(self,
+                 num_classes=None,
+                 sample_weight=None,
+                 name='MatthewsCorrelationCoefficient',
+                 dtype=tf.float32):
+        super(MatthewsCorrelationCoefficient, self).__init__(
+            name=name, dtype=dtype)
+        self.num_classes = num_classes
+        self.true_positives = self.add_weight(
+            'true_positives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+        self.false_positives = self.add_weight(
+            'false_positives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+        self.false_negatives = self.add_weight(
+            'false_negatives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+        self.true_negatives = self.add_weight(
+            'true_negatives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+
+    # TODO: sample_weights
+    def update_state(self, y_true, y_pred, sample_weight=None):
+        self.sample_weight = sample_weight
```

Comment:
Would you support `sample_weight` in this PR? If not, please remove this line and move.

Reply:
Thanks @WindQAQ. I'll temporarily remove it.